backend/server.py

```diff
@@ -40,7 +51,49 @@ def upload_file():
         
         samples = GoogleSensorParser([f.filename]).parse_files()
 
-        return samples
+        return {'type': 'upload', 'data': samples}
+
+"""Handles requests for stats data and responds with the requested
+    stats.
+
+Attributes:
+    request.data: The data sent from the frontend.
+        request.data.channels: Key value pairs of channels and arrays
+        to compute statistics for.
+"""
+@app.route('/stats', methods = ['POST'])
+def compute_stats():
+    if request.method == "POST":
+        received = json.loads(request.data)
+        avgs, stdevs = {}, {}
+
+        for i, j in enumerate(received['channels']):
+            avgs[j] = compute_running_avg(received['channels'][j])
+            stdevs[j] = compute_stdev(received['channels'][j])
+
+        return {'type': 'stats', 'avgs': avgs, 'stdevs': stdevs}
+
+"""Computes the running average of a single data trace.
+    Returns a Python list.
+
+Attributes:
+    trace: The Python list data trace to compute averages for.
+"""
+def compute_running_avg(trace):
+    n = len(trace)
+    avgs = np.cumsum(trace[:n+1], dtype=float)
+    for i in range(len(avgs)):
```

Comment:
This is cumulative average, not running average

Reply:
The running average looks at the previous k data points to determine the average. How should I determine k? Could I just take some % of the data for k?

Reply:
Rather than a % you could use 101 points. 50 on either side of the point in question.

Reply:
Rather than a %, just use last 100 points

Reply:
But then if the data trace is < 100 points, the running average would be the same as the cumulative avg right?